src/client/fuse_ll.cc

```diff
@@ -163,6 +164,61 @@ class CephFuse::Handle {
   struct fuse_args args;
 };
 
+static bool already_mounted(const char *path)
+{
+  struct stat orig_stat;
+  struct stat parent_stat;
+  char *orig_path = NULL;
+  char *parent_path = NULL;
+  bool ret_val = false;
+  int err = 0;
+
+  do
+  {
+    orig_path = new char[strlen(path) + 1];
+    memcpy(orig_path, path, strlen(path) + 1);
+    parent_path = dirname(orig_path);
+
+    // get stat information for original path
+    err = stat(orig_path, &orig_stat);
```

Comment:
Here you should use the `path` instead of the `orig_path`, from the manpage it says the `dirname()` will change or truncate the `orig_path` buffer and then return the pionter of `orig_path` ?

Reply:
Thanks :+1:  .
Done.